P4Plugin/Source/P4Task.cpp

```diff
@@ -678,13 +698,84 @@ bool P4Task::HasUnicodeNeededError( VCSStatus status )
 	return StatusContains(status, "Unicode server permits only unicode enabled clients");
 }
 
+bool P4Task::HasServerFingerPrintError(VCSStatus status)
+{
+	return StatusContains(status, "The authenticity of");
+}
+
 void P4Task::EnableUTF8Mode()
 {
 	CharSetApi::CharSet cs = CharSetApi::UTF_8;
 	m_Client.SetTrans( cs, cs, cs, cs );
 	m_Client.SetCharset("utf8");
 }
 
+#if defined(_WIN32) || defined(_WINDOWS)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#undef WIN32_LEAN_AND_MEAN
+
+bool P4Task::ShowOKCancelDialogBox(const std::string& windowTitle, const std::string& message)
+{
+	int msgBoxRet = MessageBoxW(
+		NULL,
+		(LPCWSTR)std::wstring(message.begin(), message.end()).c_str(),
+		(LPCWSTR)std::wstring(windowTitle.begin(), windowTitle.end()).c_str(),
+		MB_ICONWARNING | MB_OKCANCEL | MB_DEFBUTTON2
+	);
+
+	bool clickedOK = false;
+	if (msgBoxRet == IDOK)
+		clickedOK = true;
+
+	return clickedOK;
+}
+
+#elif defined(__linux) || defined(linux) || defined(__linux__) || defined(__gnu_linux__)
+#include <gtk/gtk.h>
+
+static bool s_IsGtkInitialized    = false;
+static bool s_FingerprintChoiceOK = false;
+
+static void HandleDialogResponseAndExitLoop(GtkWidget* widget, gint response, gpointer user_data)
+{
+	s_FingerprintChoiceOK = response == GTK_RESPONSE_ACCEPT;
+	gtk_widget_destroy(widget);
+	gtk_main_quit();
+}
+
+bool P4Task::ShowOKCancelDialogBox(const std::string& windowTitle, const std::string& message)
+{
+	s_FingerprintChoiceOK = false;
+
+	if(!s_IsGtkInitialized)
+		gtk_init(NULL, NULL);
+
+	GtkWidget *dialog = gtk_dialog_new_with_buttons(windowTitle.c_str(),
```

Comment:
What cleans up memory of Gtk `dialog`, `label` etc objects?

Reply:
Seems like deleting the dialog window should clean up its children https://people.gnome.org/~shaunm/girdoc/C/Gtk.Widget.destroy.html